src/chainstate/coordinator/mod.rs

```diff
@@ -658,6 +737,240 @@ impl<
         Ok(())
     }
 
+    /// This function reads veto-related information from the exit-at-rc clarity contract.
+    /// It returns true if the veto succeeded, and false if it failed.
+    pub fn read_veto_state(
+        &mut self,
+        rc_cycle_of_veto: u64,
+        proposed_exit_rc: u64,
+        stacks_block_id: &StacksBlockId,
+    ) -> Result<bool, Error> {
+        let reward_cycle_length = self.burnchain.pox_constants.reward_cycle_length;
+        let veto_percent_threshold = self
+            .burnchain
+            .exit_contract_constants
+            .veto_confirmation_percent;
+        let is_mainnet = self.burnchain.is_mainnet();
+        self.chain_state_db
+            .with_read_only_clarity_tx(&self.sortition_db.index_conn(), &stacks_block_id, |conn| {
+                conn.with_clarity_db_readonly(|db| {
+                    // TODO (#3034): get correct contract ID once contract is published
+                    let exit_at_rc_contract = boot_code_id("exit-at-rc", is_mainnet);
```

Comment:
Just to make sure we don't accidentally leave any of these calls to `boot_code_id()` around, could you rename this to `exit_at_reward_cycle_code_id()` or something like that and for now have it use the boot code address?  Thanks!

Reply:
Good idea!